src/Cli/Microsoft.DotNet.Cli.Utils/PathUtility.cs

```diff
@@ -97,6 +97,32 @@ public static bool TryDeleteDirectory(string directoryPath)
             }
         }
 
+        /// <summary>
+        /// Deletes the provided file. Then deletes the parent directory if empty
+        /// and continues to its parent until it fails. Returns whether it succeeded
+        /// in deleting the file it was intended to delete.
+        /// </summary>
+        public static bool DeleteFileAndEmptyParents(string path)
+        {
+            try
+            {
+                File.Delete(path);
+                var dir = Path.GetDirectoryName(path);
+
+                // Directory.Delete throws an exception when it fails to delete
+                // a directory, as, for instance, if it isn't empty. This is
+                // intended to run until it throws an exception, then return.
+                while (true)
+                {
+                    Directory.Delete(dir);
+                    dir = Path.GetDirectoryName(dir);
+                }
+            }
+            catch (Exception) { }
```

Comment:
Is there a specific exception you get for non-empty directories. If permissions are invalid, do we want to ignore those or log them to make the user aware of the underlying problem?

Reply:
I think both of those reasons are very possible. We should definitely always ignore an error due to a non-empty directory (that's an IOException). The other exception we're reasonably likely to hit (UnathorizedAccessException) I think we should ignore but mostly because this is a generic PathUtility method rather than a specific I-expect-to-be-able-to-delete-it method. If you want to make it just catch IOException, I'd probably have to handle UnauthorizedAccessException separately, but I don't mind doing that. Let me know what you think.

Reply:
You could look at using `Directory.EnumerateFileSystemEntries().Any()` - it's lazy evaluation so you shouldn't incur a huge cost (unlike `GetFileSystemEntries`). That way, any other exception, including `IOException` will likely be a valid error we'd want the calling method to report.

Reply:
I wanted to avoid doing too many expensive file system operations, but it sounds like Directory.EnumerateFileSystemEntries() is fast, so I'll use that and get rid of the try/catch. Thanks!